spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresSubscribableChannel.java

```diff
@@ -160,19 +178,29 @@ protected boolean doSend(Message<?> message, long timeout) {
 	@Override
 	public void notifyUpdate() {
 		this.executor.execute(() -> {
+			Optional<?> dispatchedMessage;
+			do {
+				dispatchedMessage = askForMessage();
+			} while (dispatchedMessage.isPresent());
+		});
+	}
+
+	private Optional<?> askForMessage() {
```

Comment:
These method names seem to be misleading; we are actually dispatching; maybe `dispatchMessageIfPresent`?